pkg/controller/inferenceservice/resources/istio/virtualservice.go

```diff
@@ -179,42 +178,86 @@ func (r *VirtualServiceBuilder) CreateVirtualService(isvc *v1alpha2.InferenceSer
 		}
 	}
 	// prepare the predict route
-	predictRoute := &istiov1alpha3.HTTPRoute{
-		Match: []*istiov1alpha3.HTTPMatchRequest{
-			{
-				Uri: &istiov1alpha3.StringMatch{
-					MatchType: &istiov1alpha3.StringMatch_Regex{
-						Regex: constants.PredictPrefix(),
+	annotations := isvc.GetAnnotations()
+	if v, ok := annotations[string(constants.CustomURLAnnotation)]; ok {
+		urls := strings.Split(v, ",")
+		for i, s := range urls {
+			log.Info("cqstest", "urls i ", i, "s", s)
+			predictRoute := &istiov1alpha3.HTTPRoute{
+				Match: []*istiov1alpha3.HTTPMatchRequest{
```

Comment:
I think this is getting a bit verbose now, can we make a function called `makeHttpRoute(prefix, serviceHostname)`?

Reply:
accept

Reply:
The `HTTPMatchRequest` can also wrapped into a function, the if/else block is mostly the same except the prefix is different. This can also be reused when creating explainer route later.

Reply:
It is a good suggestion.